### PR TITLE
Add crate name fix for webkit2xx

### PR DESCRIPTION
**Pull request by EPashkin**
web_kit2_sys and web_kit2_web_extension_sys looks ugly

### CODE REVIEW

**Comment by EPashkin** [comment]
Added config option to override crate names

**Comment by EPashkin** [comment]
@GuillaumeGomez and @sdroege: maybe there better way to do this?
And please check added text in readme.md

File: README.md [hunk 1]
```diff
@@ -435,3 +435,16 @@ unsafe { println!("{}", myUnion.test };
 ```
 
 This is required as the rust compiler can not guarantee the safety of the union, or that the member being addressed exists. The union RFC is [here](https://github.com/tbu-/rust-rfcs/blob/master/text/1444-union.md) and the tracking issue is [here](https://github.com/rust-lang/rust/issues/32836).
+
+### Crate name overrides
+
+`gir` uses simple rule to convert namespace to crate name and it sometimes go wrong, Ex. "WebKit2WebExtension" namespace converted to "web_kit2_web_extension" crate name that looks bad.
```
Comment (GuillaumeGomez) [inline README.md]: With corrections:

```
`gir` uses simple rule to convert a namespace to a crate name and it sometimes goes wrong. For example, "WebKit2WebExtension" namespace will be converted to "web_kit2_web_extension", which looks bad.
```

File: README.md [hunk 2]
```diff
@@ -435,3 +435,16 @@ unsafe { println!("{}", myUnion.test };
...
+
+To fix it table `crate_name_overrides` can be used.
```
Comment (GuillaumeGomez) [inline README.md]:
```
To fix it, the `crate_name_overrides` option can be used.
```

File: README.md [hunk 3]
```diff
@@ -435,3 +435,16 @@ unsafe { println!("{}", myUnion.test };
...
+
+This also replaces name FFI crates.
```
Comment (GuillaumeGomez) [inline README.md]:
```
This also replace FFI crates' name.
```

Comment (EPashkin) [reply README.md]: "'" really needed?

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks

**Comment by EPashkin** [comment]
Updated

File: README.md [hunk 4]
```diff
@@ -435,3 +435,16 @@ unsafe { println!("{}", myUnion.test };
...
 
 This is required as the rust compiler can not guarantee the safety of the union, or that the member being addressed exists. The union RFC is [here](https://github.com/tbu-/rust-rfcs/blob/master/text/1444-union.md) and the tracking issue is [here](https://github.com/rust-lang/rust/issues/32836).
+
+### Crate name overrides
+
+`gir` uses simple rule to convert a namespace to a crate name and it sometimes goes wrong. For example, "WebKit2WebExtension" namespace will be converted to "web_kit2_web_extension", which looks bad.
+
+To fix it, the `crate_name_overrides` option can be used.
+
+This also replace FFI crates name.
```
Comment (GuillaumeGomez) [inline README.md]: The `'` after "crates" is important. Copy and paste it. ;)

```
It also replaces FFI crates' name.
```

File: README.md (hunk 3 above)
Comment (GuillaumeGomez) [reply README.md]: Yes, it's to mark "possession". But if someone else could confirm it maybe?

Comment (EPashkin) [reply README.md]: Thanks, I didn't know that this case is possible ["We use ’s after a singular noun and ’ after a plural noun."](https://dictionary.cambridge.org/grammar/british-grammar/determiners/possession-john-s-car-a-friend-of-mine)

File: README.md (hunk 4 above)
Comment (EPashkin) [reply README.md]: Done

**Comment by GuillaumeGomez** [comment]
Looks good to me, thanks!

**Comment by EPashkin** [comment]
Ok, lets merge then

**Pull request merged**